### ISSUE TITLE
Unable to install node-red-contrib-composer Error: stderr maxBuffer exceeded

### ISSUE THREAD

**Issue by davidkel** [node-red/node-red#1488]
Our module does generate an awful lot of warnings when it installs because it does native compilations so I wonder of the stdErr buffer is being exceeded with all the noise that trying to install this component generates.
The obvious workaround is to install manually of course, but thought that maybe something that could be done to allow it to be installed through the UI ?

### What are the steps to reproduce?
Run the node red docker container, using the browser to manage the palette try to install 
Installing module: node-red-contrib-composer, version: 0.0.9

### What happens?
The following is output
```
24 Nov 12:31:57 - [info] Installing module: node-red-contrib-composer, version: 0.0.9
24 Nov 12:32:35 - [warn] Installation of module node-red-contrib-composer failed:
24 Nov 12:32:35 - [warn] ------------------------------------------
24 Nov 12:32:35 - [warn] Error: stderr maxBuffer exceeded
24 Nov 12:32:35 - [warn] ------------------------------------------
```

### What do you expect to happen?
It should install and I can install it in the container using npm install.
### Please tell us about your environment:

- [ ] Node-RED version: v0.17.5
- [ ] node.js version: v6.12.0
- [ ] npm version: 3.10.10
- [ ] Platform/OS: node red docker image
- [ ] Browser: chrome

**Comment by knolleary** [comment]
Thanks for raising this - agreed we shouldn't fall over on this.

Need to either increase the buffer size used by `child_process.exec` or move over to `child_process.spawn`. Either way, need to think about how to best log the output from npm to help the user diagnose failures better.

**Comment by knolleary** [comment]
fixed in 0.18 branch
